bin.sh

```diff
@@ -1,32 +1,73 @@
 #!/bin/bash
 
+# Default parameters
+ALLOW_SELF_REGISTER="true"
+IPV4_RANGES=""
+IPV6_RANGES=""
+ALLOWED_SIGN_UP_EMAIL_DOMAINS=""
+
+# Parse command-line arguments for customization
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --disable-self-register) ALLOW_SELF_REGISTER="false" ;;
+        --ipv4-ranges) IPV4_RANGES="$2"; shift ;;
+        --ipv6-ranges) IPV6_RANGES="$2"; shift ;;
+        --allowed-signup-email-domains) ALLOWED_SIGN_UP_EMAIL_DOMAINS="$2"; shift ;;
+        *) echo "Unknown parameter: $1"; exit 1 ;;
+    esac
+    shift
+done
+
+
+# Validate the template
+aws cloudformation validate-template --template-body file://deploy.yml  > /dev/null 2>&1
+if [[ $? -ne 0 ]]; then
+    echo "Template validation failed"
+    exit 1
+fi
+
 StackName="CodeBuildForDeploy"
 
-stackId=$(aws cloudformation create-stack \
+# Deploy the CloudFormation stack
+aws cloudformation deploy \
   --stack-name $StackName \
-  --template-body file://deploy.yml \
+  --template-file deploy.yml \
   --capabilities CAPABILITY_IAM \
-  --query 'StackId' --output text)
+  --parameter-overrides AllowSelfRegister=$ALLOW_SELF_REGISTER Ipv4Ranges="$IPV4_RANGES" Ipv6Ranges="$IPV6_RANGES" AllowedSignUpEmailDomains="$ALLOWED_SIGN_UP_EMAIL_DOMAINS"
```

Comment:
Requires logic convertion between DISABLE_SELF_REGISTER and `AllowSelfRegister`?

Reply:
No need to, the var set is ALLOW_SELF_REGISTER